Various stuffs

WinDanesz opened the pull request.

Commit:
added a big bunch of sfx files, assigned sounds for all entities

Commit:
optimized all entity SFX for smaller size

P3pp3rF1y commented on src/main/java/net/shadowmage/ancientwarfare/npc/ai/NpcAIAttackMeleeLongRange.java:
```diff
@@ -37,6 +38,9 @@ protected void doAttack(double distanceToEntity) {
 			npc.attackEntityAsMob(getTarget());
 			setAttackDelay(getCoolDown());
 			npc.addExperience(AWNPCStatics.npcXpFromAttack);
+			if ((Math.random() < 0.2 && npc instanceof NpcFaction)) {
```
We should play the sounds in attackEntityAsMob rather than AI. that should also allow you to override attackEntityAsMob and call the play sound in addition to calling super.attackEntityAsMob and there won't be a need for instanceof check in that case

P3pp3rF1y commented on src/main/java/net/shadowmage/ancientwarfare/npc/entity/faction/NpcFaction.java:
```diff
@@ -249,6 +252,39 @@ public void onEntityUpdate() {
+	public void playAttackSound() {
+		String sound = getEntitySound() + "_attack";
+		if (!sound.equals("generic_attack") && AWNPCSounds.isValidSound(sound)) {
+			playSound(AWNPCSounds.getSoundEventFromString(sound), 1.0F, 1.2F / (rand.nextFloat() * 0.2F + 0.9F));
```
The sound event should be cached in a field so that there isn't a need to query to compile its id and query for it every time a sound needs to be played. Probably best way to approach that is compile the name in constructor and cache it into something like private final SoundEvent attackSound;

P3pp3rF1y commented on src/main/java/net/shadowmage/ancientwarfare/npc/init/AWNPCSounds.java:
```diff
@@ -7,11 +7,130 @@
+
+	public static Map<String, SoundEvent> npcSoundEvents = new HashMap<>();
+
+	public static final SoundEvent ENTITY_BARBARIAN_ATTACK = InjectionTools.nullValue();
```
I feel like all this ENTITY_ prefixing is just quite a bit of additional noise. Vanilla may do it, but it may be easier on eyes to just remove it. The same applies to the event names in the sounds.json. So unless you feel strongly otherwise let's remove these prefixes.

Commit:
removed ENTITY_ prefixes from entity sound files

Commit:
moved the faction entity SoundEvents to constant fields, moved playAttackSound call to attackEntityAsMob

WinDanesz commented:
made the changes

Commit:
Added hurt and death sounds for player owned NPCs

WinDanesz commented:
also added hurt and death sounds for player owned NPCs

Commit:
Added  a check for Alex model too because the player owned ones are males by default

Commit:
Fixed mailbox private and auto output checkboxes

Commit:
Increased gates' health from 40 to 80

Commit:
Added Gate/Drawbridge moving, hurt, break sound,
Changed Gate/Drawbridge resistance against tools and weapons

WinDanesz commented:
Fixed mailbox private and auto output checkboxes
Increased Gate/Drawbridge health from 40 to 80
Added Gate/Drawbridge moving, hurt, break sound
Changed Gate/Drawbridge resistance against tools and weapons

Commit:
fixed some sound entries

Commit:
Added new Auto Crafting station (model by Silentine)

WinDanesz commented:
fixed some sound entries

WinDanesz commented:
Added new Auto Crafting station (model by Silentine)

Commit:
Fixed player owned workers flee from the player's tamed mobs

WinDanesz commented:
Fixed player owned workers flee from the player's tamed mobs
